src/transformStateWithClones.js

```diff
@@ -1,13 +1,40 @@
 'use strict';
-
-/**
+/*
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const updateState = { ...state };
+  const updateActions = [];
+
+  for (const key of actions) {
+    switch (key.type) {
+      case 'addProperties':
+        Object.assign(updateState, key.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const item of key.keysToRemove) {
```

Comment:
What item is it? item/el/elem/arr/array/obj/object/res/result and etc are bad names. Name should describe a variable